internal/util/debug/debug.go

```diff
@@ -97,25 +123,98 @@ func Listen(opts *ListenOpts) (*Handler, error) {
 		}),
 	))
 
+	http.HandleFunc("/debug/archive", promhttp.InstrumentHandlerDuration(
+		archiveDurations,
+		http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Set("Content-Type", "application/zip")
+
+			rw.Header().Set(
+				"Content-Disposition",
+				fmt.Sprintf("attachment; filename=FerretDB-debug-%d.zip", time.Now().UnixMilli()),
+			)
+
+			ctx := req.Context()
+			zipWriter := zip.NewWriter(rw)
+
+			defer func() {
+				if err := zipWriter.Close(); err != nil {
+					opts.L.Error("Archive handler failed", zap.Error(err))
+					rw.WriteHeader(http.StatusInternalServerError)
```

Comment:
It is not possible to send a header with status after some content was already sent

Reply:
My bad, fixed